lib/ansible/modules/cloud/amazon/cloudformation_stack_set.py

```diff
@@ -0,0 +1,664 @@
+                    "`template_body`, or `template_url`".format(module.params['name'])
+            )
+
+    stack_params['Parameters'] = []
+    for k, v in module.params['parameters'].items():
+        if isinstance(v, dict):
+            # set parameter based on a dict to allow additional CFN Parameter Attributes
+            param = dict(ParameterKey=k)
+
+            if 'value' in v:
+                param['ParameterValue'] = to_native(v['value'])
+
+            if 'use_previous_value' in v and bool(v['use_previous_value']):
+                param['UsePreviousValue'] = True
+                param.pop('ParameterValue', None)
+
+            stack_params['Parameters'].append(param)
+        else:
+            # allow default k/v configuration to set a template parameter
+            stack_params['Parameters'].append({'ParameterKey': k, 'ParameterValue': str(v)})
+
+    if module.params.get('tags') and isinstance(module.params.get('tags'), dict):
+        stack_params['Tags'] = ansible.module_utils.ec2.ansible_dict_to_boto3_tag_list(module.params['tags'])
+
+    if module.params.get('administration_role_arn'):
+        # TODO loosen the semantics here to autodetect the account ID and build the ARN
+        stack_params['AdministrationRoleARN'] = module.params['administration_role_arn']
+    if module.params.get('execution_role_name'):
+        stack_params['ExecutionRoleName'] = module.params['execution_role_name']
+
+    result = {}
+
+    if module.check_mode:
+        if state == 'absent' and existing_stack_set:
+            module.exit_json(changed=True, msg='Stack set would be deleted', meta=[])
+        elif state == 'absent' and not existing_stack_set:
+            module.exit_json(changed=False, msg='Stack set doesn\'t exist', meta=[])
+        elif state == 'present' and not existing_stack_set:
+            module.exit_json(changed=True, msg='New stack set would be created', meta=[])
+        elif state == 'present' and existing_stack_set:
+            new_stacks, existing_stacks, unspecified_stacks = compare_stack_instances(
+                module,
+                cfn,
+                module.params['name'],
+                module.params['accounts'],
+                module.params['regions'],
+            )
+            if new_stacks:
+                module.exit_json(changed=True, msg='New stack instance(s) would be created', meta=[])
+            elif unspecified_stacks and module.params.get('purge_stack_instances'):
+                module.exit_json(changed=True, msg='Old stack instance(s) would be deleted', meta=[])
+        else:
+            # TODO: need to check the template and other settings for correct check mode
+            module.exit_json(changed=False, msg='No changes detected', meta=[])
+
+    if state == 'present':
+        if not existing_stack_set:
+            # on create this parameter has a different name, and cannot be referenced later in the job log
```

Comment:
This code path should set result['changed'] = True, same for creating new stack instances.